pkgs/development/libraries/freetype/default.nix

```diff
@@ -62,6 +62,9 @@ in stdenv.mkDerivation rec {
   doCheck = true;
 
   postInstall = glib.flattenInclude + ''
+    substituteInPlace $dev/bin/freetype-config \
+      --replace ${buildPackages.pkgconfig} ${pkgconfig}
```

Comment:
Is this correct? I would think you would want buildPackages pkgconfig in this case.

Reply:
I don't think so `freetype-config` is build to the host machine and buildPackages.pkgconfig has the wrong architecture. We also have a check for that to warn about those cases.

Reply:
Dealing with `-config` style applications is a different challenge we have to face. This would need some kind of `buildPackage` that points to the host packages.

Reply:
Yeah I would straight up
```nix
rm "$dev/bin"
cp -r "${buildPackages.freetype.dev}/bin" "$dev/"
```
and file an issue upstream with the maintainer so we can easily build `*-config` as a separate derivation and sidestep the issue entirely.

`*-config` style stuff sucks for the same reason compilers insisting on building their own standard libraries sucks. Packages' installations should be for only one platform (/ not cross bootstrapping stages).

Reply:
Oh wait, I forgot that the `*-config` binaries are hard-coded with that necessary data often, so my `cp -r` won't work. Ugh!

Reply:
Thanks.